### PR TITLE
search: test sort, facet, pagination endpoints [+]

### LINKED ISSUE #56: Make maximum number of results configurable
(maybe this doesn't need to be configurable if ES caps at 10000)

This could be configured in invenio-records-rest, so maybe we want to be able to configure this still.

(honestly I am not a particularly big proponent of this)

### PR DESCRIPTION
**Pull request by fenekku**
- Service level defaults are used for pagination
- Pagination errors are dealt at the service level
- Link generation is pushed off to another issue.
- closes #56

### CODE REVIEW

File: invenio_records_resources/services/records/params/pagination.py [hunk 1]
```diff
@@ -17,12 +20,22 @@ class PaginationParam(ParamInterpreter):
 
     def apply(self, identity, search, params):
         """Evaluate the query str on the search."""
+        options = self.config.search_pagination_options
+
+        default_size = options["default_results_per_page"]
+        default_max_results = options["default_max_results"]
+
+        params.setdefault("size", default_size)
```
Comment (lnielsen) [inline invenio_records_resources/services/records/params/pagination.py:28]: Comment: I like this much better than the previous approach - much cleaner in IMHO 👍

File: invenio_records_resources/resources/errors.py [hunk 2]
```diff
@@ -40,3 +41,11 @@ def pid_redirected_error_handler(e):
             raise e
 
     return pid_redirected_error_handler
+
+
+handle_querystring_validation_error = create_errormap_handler(
```
[the remark this answers is not in the document]
Comment (fenekku) [reply invenio_records_resources/resources/errors.py:46]: Will move it to records_config :+1:

File: invenio_records_resources/services/records/config.py [hunk 3]
```diff
@@ -38,24 +38,37 @@ class RecordServiceConfig(ServiceConfig):
...
             title=_('Best match'),
             fields=['_score'],  # ES defaults to desc on `_score` field
         ),
-        mostrecent=dict(
+        "-bestmatch": dict(
+            title=_('Worst match'),
+            fields=['-_score'],
+        ),
+        "mostrecent": dict(
             title=_('Most recent'),
```
[the remark this answers is not in the document]
Comment (fenekku) [reply invenio_records_resources/services/records/config.py:56]: :heavy_check_mark:

File: invenio_records_resources/services/records/config.py [hunk 4]
```diff
@@ -38,24 +38,37 @@ class RecordServiceConfig(ServiceConfig):
...
+            fields=['created'],
+        ),
+    }
     search_facets_options = dict(
         aggs={},
         post_filters={}
     )
+    search_pagination_options = {
+        "default_results_per_page": 25,
+        "default_max_results": 10000
```
[the remark this answers is not in the document]
Comment (fenekku) [reply invenio_records_resources/services/records/config.py:70]: I see, max results isn't really configurable(?)

File: tests/services/test_service_pagination.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (fenekku) [reply tests/services/test_service_pagination.py:1]: Yes I saw that in invenio-records-rest and kept the same functionality even though it was strange! Good to know I can fix it. We will only have a `next` link if the next link does exist.

File: invenio_records_resources/services/records/params/pagination.py (hunk 1 above)
Comment (fenekku) [reply invenio_records_resources/services/records/params/pagination.py:28]: Ah good! I know there are other ways to go about it, but I thought this one carried the clearest meaning.

File: tests/services/test_service_pagination.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (fenekku) [reply tests/services/test_service_pagination.py:1]: I will take care of this in the other PR

**Comment by fenekku** [comment]
Ready to be re-reviewed. @lnielsen @ppanero @zzacharo

**Pull request merged**